analysis/02_new_model_input/03.1_create_rainfall_dataset.md

```diff
@@ -0,0 +1,143 @@
+
+# grid
+grid_name = "analysis/02_new_model_input/02_housing_damage/output/phl_0.1_degree_grid_land_overlap.gpkg"
+grid_file = Path(os.getenv("STORM_DATA_DIR")) / grid_name
+grid = gpd.read_file(grid_file)
+
+# gpm data
+gpm_file_name = "gpm_data/rainfall_data/output_hhr/"
+gpm_folder_path = os.path.join(input_dir, gpm_file_name)
+typhoon_list = os.listdir(gpm_folder_path)
+
+# outputs
+processed_output_dir = os.path.join(
+    input_dir, "gpm_data/rainfall_data/output_hhr_processed/"
+)
+output_dir = (
+    Path(os.getenv("STORM_DATA_DIR"))
+    / "analysis/02_new_model_input/03_rainfall/output"
+)
+```
+
+Testing raster method
+
+```python
+# Testing the raster stats method
+day_list = os.listdir(gpm_folder_path + typhoon_list[1] + "/GPM/")
+file_list = os.listdir(
+    gpm_folder_path + typhoon_list[1] + "/GPM/" + day_list[0]
+)
+file = Path(
+    gpm_folder_path
+    + typhoon_list[1]
+    + "/GPM/"
+    + day_list[0]
+    + "/"
+    + file_list[0]
+)
+input_raster = rasterio.open(file)
+array = input_raster.read(1)
+# checking if crs are the same
+input_raster.crs == grid.crs
+```
+
+```python
+# computing stats for 4 adjacent cells
+summary_stats = zonal_stats(
+    grid,
+    array,
+    stats=["min", "max", "mean", "std"],
+    nodata=29999,
+    all_touched=True,
+    affine=input_raster.transform,
+)
+grid_stats = pd.DataFrame(summary_stats)
+```
+
+Looping over typhoons
+
+This section takes a very long time to process.
+
+```python
+# setting up loop for running through all typhoons
+# extracting the max and mean of the 4 adjacent cells due to shifting to grids
+stats_list = ["mean", "max"]
+for typ in typhoon_list:
+    day_list = os.listdir(gpm_folder_path + typ + "/GPM/")
+    day_df = pd.DataFrame()
+    for day in day_list:
+        file_list = os.listdir(gpm_folder_path + typ + "/GPM/" + day)
+        file_df = pd.DataFrame()
+        for file in file_list:
+            if file.startswith("3B-HHR"):
+                file_path = Path(
+                    gpm_folder_path + typ + "/GPM/" + day + "/" + file
+                )
+                input_raster = rasterio.open(file_path)
+                array = input_raster.read(1)
+                summary_stats = zonal_stats(
+                    grid,
+                    array,
+                    stats=stats_list,
+                    nodata=29999,
+                    all_touched=True,
+                    affine=input_raster.transform,
+                )
+                grid_stats = pd.DataFrame(summary_stats)
+                # change values by dividing by 10 to mm/hr
+                grid_stats[stats_list] /= 10
+                grid_merged = pd.merge(
+                    grid.drop(["geometry", "Longitude", "Latitude"], axis=1),
+                    grid_stats,
+                    left_index=True,
+                    right_index=True,
+                )
+                grid_merged["start"] = str(
+                    file[35:37] + ":" + file[37:39] + ":" + file[39:41]
```

Comment:
Where do these indices come from,  is it the filename construction? Is there a more intuitive way to do it?